### DISCUSSION TITLE
Variable scoping in x-for

### DISCUSSION THREAD

**Discussion by docdunning** [alpinejs/alpine#4002] [1. Help] [answered]
In the following code, the variable `person` is used as the array element alias in the x-for loop; and is also part of the x-data scope.

I would expect that the `person` in the x-for loop is scoped locally to that; and that it isn't the same as the `person` in the x-data scope.

OK, I know this isn't good practice, and it's easily avoided.  

My question, though, is whether there is a risk that Alpine itself could somehow use the same variable and end up getting the two uses confused.

```
<div x-data="{
    people: [
        {'id':1, name:'James'},
        {'id':2, name:'Alice'},
        {'id':3, name:'Henry'},
    ],
    person: {'id':10, name:'Billy'},
    }"
>

<template x-for="person in people">
    <p x-text="person.name"></p>
</template>

<p x-text="person.name"></p>

</div>
```

**Comment by ekwoka** [comment, marked as answer]
It will not have any concern getting confused.

A simple mental model of how it works is that 

at the `p[x-text]` it goes "Okay, we want `person`, lets walk up the tree until it's defined. It's not defined on a `p` tag, okay, it's defined on the `template` as this object, there that's person".

(this is technically incorrect since the template defines the values on the child but it works better as a mental model).

There isn't any way for it to get confused, only you as the developer moving elements around and not realizing you changed scopes.

**Comment by docdunning** [comment]
@ekwoka 

> .. only you as the developer moving elements around and not realizing you changed scopes.

Which is more than likely ;-)

Thanks for explaining.
